solidity/test/ERC20NoData.ts

```diff
@@ -0,0 +1,201 @@
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { ERC20NoData } from '../typechain';
+
+describe('ERC20NoData - Unit Tests', function () {
+  const contractName = 'testName';
+  const contractSymbol = 'testSymbol';
+  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+  const ONE_ADDRESS = '0x1111111111111111111111111111111111111111';
+  let deployedERC20NoData: ERC20NoData;
+  let Factory;
+
+  let deployerSignerA: SignerWithAddress;
+  let signerB: SignerWithAddress;
+  let signerC: SignerWithAddress;
+
+  beforeEach(async () => {
+    [deployerSignerA, signerB, signerC] = await ethers.getSigners();
+    Factory = await ethers.getContractFactory('ERC20NoData');
+    // Deploy erc20 token pool with Signer A
+    deployedERC20NoData = await Factory.connect(deployerSignerA).deploy(
+      contractName,
+      contractSymbol,
+    );
+    await deployedERC20NoData.deployed();
+  });
+
+  it('Create - Should create a new ERC20 instance with default state', async function () {
+    expect(await deployedERC20NoData.name()).to.equal(contractName);
+    expect(await deployedERC20NoData.symbol()).to.equal(contractSymbol);
+  });
+
+  it('Mint - Deployer should mint tokens to itself successfully', async function () {
+    expect(await deployedERC20NoData.balanceOf(deployerSignerA.address)).to.equal(0);
+    // Signer A mint to Signer A (Allowed)
+    await expect(
+      deployedERC20NoData.connect(deployerSignerA).mintNoData(deployerSignerA.address, 20),
+    )
+      .to.emit(deployedERC20NoData, 'Transfer')
+      .withArgs(ZERO_ADDRESS, deployerSignerA.address, 20);
+
+    expect(await deployedERC20NoData.balanceOf(deployerSignerA.address)).to.equal(20);
+  });
+
+  it('Mint - Non-deployer of contract should not be able to mint tokens', async function () {
+    expect(await deployedERC20NoData.balanceOf(signerB.address)).to.equal(0);
+    // Signer B mint to Signer B (Not allowed)
+    await expect(
+      deployedERC20NoData.connect(signerB).mintNoData(signerB.address, 20),
+    ).to.be.revertedWith('Ownable: caller is not the owner');
+
+    expect(await deployedERC20NoData.balanceOf(signerB.address)).to.equal(0);
+  });
+
+  it('Mint - Non-signing address should not be able to mint tokens', async function () {
+    expect(await deployedERC20NoData.balanceOf(ONE_ADDRESS)).to.equal(0);
+    // Non-signer mint to non-signer (Not allowed)
+    await expect(deployedERC20NoData.connect(ONE_ADDRESS).mintNoData(ONE_ADDRESS, 20)).to.be
```

Comment:
may need to check the revert message here, it may have failed because the ONE_ADDRESS is not the contract owner?

Reply:
`Error: VoidSigner cannot sign transactions` is the error thrown.